### ISSUE TITLE
add a context manager to easily "freeze" excel when running code

### ISSUE THREAD

**Issue by sdementen** [xlwings/xlwings#254]
It is often useful to freeze excel while running some code, freeze in the sense of disabling (=setting to False):
- Application.ScreenUpdating
- Application.EnableEvents
- Application.DisplayAlerts
- setting calculation mode to manual

A nice API would be a simple context_manager in the vein of:

```
with app.freeze(events=True, screen=True, alerts=True, calculation=True):
    ... do some expensive changes in a workbook ...
```

that would save the status of these variables, change them according to the flags, yield and then set back the variables back to their original status.

Any feedback on the API ? if ok, i'll do a PR for the Windows version (I have no experience nor access to a Mac)

**Comment by fzumstein** [comment]
This seems like a good idea, and the API can still be discussed/fine tuned before the first release...

**Comment by zogzog** [comment]
app.interact could be slightly less confusing than app.freeze but I like the idea.

**Comment by sdementen** [comment]
What about 
```
with app.enable(events=False, screen=True, alerts=False, calculation=False):
    ... do some expensive changes in a workbook ...
    ... events, alerts and calculation are disable/frozen
```
We could also add the Application.Interactive attribute as app.interactive to come to for example:
```
with app.enable(calculation=True, interactive=False):
    ... do some expensive changes in a workbook ...
```

**Comment by fzumstein** [comment]
implemented in #1671
